include/pybind11/attr.h

```diff
@@ -345,9 +345,10 @@ struct type_record {
 
         bases.append((PyObject *) base_info->type);
 
-        if (base_info->type->tp_dictoffset != 0) {
-            dynamic_attr = true;
-        }
+        dynamic_attr |= base_info->type->tp_dictoffset != 0;
```

Comment:
@henryiii I am not sure if this should be an exclusive if/else . This seems to be the simplest delta though.

Reply:
other option do an #IF and #ELSE

Reply:
Seems like it's better as an `#else` for the if below?

Reply:
If `base_info->type->tp_dictoffset` is still important in 3.11 I think this is best. If not, it could be placed in an `#else`.